File: 772/solution.cpp
#include<iostream>
using namespace std;
const int mod = 1e9+7;
int add(int a, int b) { return (a+b)% mod; }
int mul(int a, int b) { return 1ll*a*b%mod; }
const int upper = 1e8;
bool p[upper];
int cal(long long ind) {
	long long buffer = ind;
	while(buffer <= upper) buffer *= ind;
	buffer /= ind;
	return buffer;
}
int main() {
	int ans = 1;
	for(int i = 2;i<upper;i++) {
		if(!p[i]) {
			ans = mul(ans, cal(i));
			for(int j = i<<1;j<upper;j+=i) p[j] = true;
		}
	}
	cout<<mul(ans,2)<<endl;
	return 0;
}

File: 772/solution.go
package main

import "fmt"

var mod = int64(1e9+7)

func add(a, b int64) int64 {
	return (a+b)%mod
}

func mul(a, b int64) int64 {
	return a*b%mod
}

var upper = int64(1e8)

var prime  = []bool{}

func cal(ind int64) int64 {
	buffer := ind
	for buffer <= upper {
		buffer *= ind
	}
	buffer /= ind
	return buffer
}

func main() {
	ans := int64(1)
	for i:=int64(0);i<upper;i++ {
		prime = append(prime, false)
	}
	for i:=int64(2);i<upper;i++ {
		if(!prime[i]) {
			ans = mul(ans, cal(i))
			for j:=i*2;j<upper;j+=i {
				prime[j] = true
			}
		}
	}
	fmt.Printf("%d\n", mul(ans, 2))
}
